runtime/src/test_utils.rs

```diff
@@ -862,6 +890,53 @@ impl<BS: Blockstore> Runtime for MockRuntime<BS> {
                 self.message().caller(), &addrs
         ))
     }
+
+    fn validate_immediate_caller_namespace<I>(&mut self, namespaces: I) -> Result<(), ActorError>
+    where
+        I: IntoIterator<Item = u64>,
+    {
+        self.require_in_call();
+
+        let namespaces: Vec<u64> = namespaces.into_iter().collect();
+
+        let mut expectations = self.expectations.borrow_mut();
+        assert!(
+            expectations.expect_validate_caller_f4_namespace.is_some(),
+            "unexpected validate caller namespace"
+        );
+
+        let expected_namespaces =
+            expectations.expect_validate_caller_f4_namespace.as_ref().unwrap();
+
+        assert_eq!(
+            &namespaces, expected_namespaces,
+            "unexpected validate caller namespace {:?}, expected {:?}",
+            namespaces, &expectations.expect_validate_caller_f4_namespace
+        );
+
+        let caller_f4 = self.lookup_delegated_address(self.caller().id().unwrap());
+
+        assert!(caller_f4.is_some(), "unexpected caller doesn't have a delegated address");
+
+        for id in namespaces.iter() {
+            let bound_address = match caller_f4.unwrap().payload() {
+                Payload::Delegated(d) => d.namespace(),
+                _ => unreachable!(
+                    "lookup_delegated_address should always return a delegated address"
+                ),
+            };
+            if bound_address == *id {
+                expectations.expect_validate_caller_f4_namespace = None;
+                return Ok(());
+            }
+        }
+        expectations.expect_validate_caller_addr = None;
+        Err(actor_error!(forbidden;
+                "caller address {:?} forbidden, allowed: {:?}",
+                self.message().caller(), &namespaces
+        ))
+    }
```

Comment:
This code shouldn't really be in a mock. The expectation object should just contain the result of the call so it's easy for the tester to engineer whichever outcome they want. Non-blocking I guess.

Reply:
I agree, though it does mirror the pattern we have with `validate_immediate_caller_is` above.